web/src/features/Dashboard/components/Dashboard.tsx

```diff
@@ -622,66 +623,34 @@ const Dashboard = (props: Props) => {
     });
   };
 
-  const getAppDashboard = (): Promise<void> => {
-    return new Promise((resolve, reject) => {
-      // this function is in a repeating callback that terminates when
-      // the promise is resolved
-
-      // TODO: use react-query to refetch this instead of the custom repeater
-      if (!props.app) {
-        resolve();
-      }
+  const { data: selectedAppClusterDashboardResponse } =
+    useSelectedAppClusterDashboardWithIntercept({ refetchInterval: 2000 });
 
-      if (props.cluster?.id === "" && props.isHelmManaged === true) {
-        // TODO: use a callback to update the state in the parent component
-        props.cluster.id = 0;
-      }
+  useEffect(() => {
+    if (selectedAppClusterDashboardResponse) {
+      setState({
+        dashboard: {
+          appStatus: selectedAppClusterDashboardResponse.appStatus,
+          prometheusAddress:
+            selectedAppClusterDashboardResponse.prometheusAddress,
+          metrics: selectedAppClusterDashboardResponse.metrics,
+        },
+      });
+    }
+  }, [selectedAppClusterDashboardResponse]);
```

Comment:
Pretty common pattern at this point:
- get rid of the fetching function
- replace with a custom hook that fetches
- set the state in `useEffect`

Eventually we will get rid of these `useEffects` and use the data directly but it's easier to use the existing state structure for this change for now